android/src/main/java/io/mosip/tuvali/verifier/Verifier.kt

```diff
@@ -1,264 +1,91 @@
-import io.mosip.tuvali.cryptography.VerifierCryptoBoxBuilder
-import io.mosip.tuvali.exception.BLEException
-import io.mosip.tuvali.openid4vpble.EventEmitter
-import io.mosip.tuvali.openid4vpble.events.withArgs.DataReceivedEvent
-import io.mosip.tuvali.openid4vpble.events.withoutArgs.ConnectedEvent
-import io.mosip.tuvali.openid4vpble.events.withoutArgs.DisconnectedEvent
-import io.mosip.tuvali.openid4vpble.events.withoutArgs.SecureChannelEstablishedEvent
-import io.mosip.tuvali.transfer.ByteCount.FourBytes
-import io.mosip.tuvali.transfer.TransferReportRequest
-import io.mosip.tuvali.transfer.Util
+import io.mosip.tuvali.common.events.DisconnectedEvent
+import io.mosip.tuvali.common.events.Event
+import io.mosip.tuvali.common.events.EventEmitter
+import io.mosip.tuvali.common.events.VerificationStatusEvent
+import io.mosip.tuvali.common.safeExecute.TryExecuteSync
+import io.mosip.tuvali.common.uri.OpenId4vpURI
+import io.mosip.tuvali.exception.handlers.ExceptionHandler
 import io.mosip.tuvali.transfer.Util.Companion.getLogTag
-import io.mosip.tuvali.verifier.exception.UnsupportedMTUSizeException
-import io.mosip.tuvali.verifier.exception.VerifierException
-import io.mosip.tuvali.verifier.transfer.ITransferListener
-import io.mosip.tuvali.verifier.transfer.TransferHandler
-import io.mosip.tuvali.verifier.transfer.message.RemoteRequestedTransferReportMessage
-import io.mosip.tuvali.verifier.transfer.message.ResponseChunkReceivedMessage
-import io.mosip.tuvali.verifier.transfer.message.ResponseSizeReadSuccessMessage
 import org.bouncycastle.util.encoders.Hex
-import java.security.SecureRandom
-import java.util.*
 
-private const val MIN_MTU_REQUIRED = 64
-
-class Verifier(
-  context: Context,
-  private val eventEmitter: EventEmitter,
-  private val handleException: (BLEException) -> Unit
-) :
-  IPeripheralListener, ITransferListener {
-  private var secretsTranslator: SecretsTranslator? = null
+class Verifier(private val context: Context): IVerifier {
   private val logTag = getLogTag(javaClass.simpleName)
-  var publicKey: ByteArray = byteArrayOf()
-  private lateinit var walletPubKey: ByteArray
-  private lateinit var nonce: ByteArray
-  private var secureRandom: SecureRandom = SecureRandom()
-  private var verifierCryptoBox: VerifierCryptoBox = VerifierCryptoBoxBuilder.build(secureRandom)
-  private var peripheral: Peripheral
-  private var transferHandler: TransferHandler
-  private val handlerThread = HandlerThread("TransferHandlerThread", THREAD_PRIORITY_DEFAULT)
-  private var maxDataBytes = 20
-  //TODO: Update UUIDs as per specification
-  companion object {
-    val SERVICE_UUID: UUID = UUID.fromString("0000AB29-0000-1000-8000-00805f9b34fb")
-    val SCAN_RESPONSE_SERVICE_UUID: UUID = UUID.fromString("0000AB2A-0000-1000-8000-00805f9b34fb")
-    const val DISCONNECT_STATUS = 1
-  }
+  private var communicator: VerifierBleCommunicator? = null
+  private var eventEmitter: EventEmitter = EventEmitter()
```

Comment:
```suggestion
  private var eventEmitter = EventEmitter()
```

(nit) Explicit type declaration might not be required for this one as this is not an interface class.

Reply:
updated